server/src/services/semanticHighlight/highlighters/StructDefinitionHighlighter.ts

```diff
@@ -0,0 +1,21 @@
+import { SemanticTokenTypes } from "vscode-languageserver-protocol";
+import { NodeType } from "@nomicfoundation/slang/cst";
+import { RuleKind, TokenKind } from "@nomicfoundation/slang/kinds";
+import { HighlightVisitor } from "../HighlightVisitor";
+import { SlangNodeWrapper } from "../../../parser/slangHelpers";
+
+// Highlights struct definitions
+export class StructDefinitionHighlighter extends HighlightVisitor {
+  public tokenKinds = new Set([TokenKind.Identifier]);
+
+  public enter(nodeWrapper: SlangNodeWrapper): void {
+    const ancestors = nodeWrapper.pathRuleNodes;
+    if (
+      nodeWrapper.type === NodeType.Token &&
```

Comment:
cc @AntonyBlakey for providing binding info from Slang.

Reply:
It's highlighting the definition and usage for me:
![image](https://github.com/NomicFoundation/hardhat-vscode/assets/2818438/140a44f0-c897-46e8-a683-a67f35750dca)
![image](https://github.com/NomicFoundation/hardhat-vscode/assets/2818438/4b274e8b-786b-41e2-994b-aa0501126211)

Reply:
Aha! I was thrown off by parameter names being highlighted the same as their types, but only in their declaration, not usage. Looks like it exists already in production though. I wonder if it can be fixed by us highlighting the parameter names?

<img width="321" alt="image" src="https://github.com/NomicFoundation/hardhat-vscode/assets/15987992/c957b640-0420-4b25-b108-02b3e2625cfc">

Reply:
You are right, it's from the textmate grammar. On the screenshot I sent, I had the grammar turned off so I could only see the effects of the semantic highlighting

Reply:
Thanks.